### PR TITLE
Fix a few cursor bugs!

### PR DESCRIPTION
**Pull request by Joshua-Ashton**
@DadSchoorse

### CODE REVIEW

**Comment by DadSchoorse** [comment]
Vulkan srgb is really annoying sometimes. Fwiw, in my fsr branch I have https://github.com/DadSchoorse/gamescope/commit/3fe231bf677f639989524c1fd7b5c4fc2cc96e65. That would avoid us having to care about srgb modifiers at all, because the primary format is always non srgb. But somehow it causes validation errors which can't really explain.

File: src/rendervulkan.cpp [hunk 1]
```diff
@@ -759,92 +759,101 @@ CVulkanTexture::~CVulkanTexture( void )
...
-		imageFormatInfo.flags = 0;
-		VkImageFormatProperties2 imageFormatProps = {};
-		imageFormatProps.sType = VK_STRUCTURE_TYPE_IMAGE_FORMAT_PROPERTIES_2;
-		VkResult res = vkGetPhysicalDeviceImageFormatProperties2( physicalDevice, &imageFormatInfo, &imageFormatProps );
-		if ( res == VK_ERROR_FORMAT_NOT_SUPPORTED )
-		{
-			continue;
-		}
-		else if ( res != VK_SUCCESS )
+	sampledShmFormats.push_back( drmFormat );
```
Comment (DadSchoorse) [inline src/rendervulkan.cpp:785]: We get duplicates here (one for srgb, one for unorm), does that matter?

Comment (Joshua-Ashton) [reply src/rendervulkan.cpp:785]: It shouldn't, but thanks for pointing it out. I will fix it when I wake up... Same for sampledDRMFormats

File: src/steamcompmgr.cpp [hunk 2]
```diff
@@ -1016,14 +1016,15 @@ bool MouseCursor::getTexture()
 	}
 
 	CVulkanTexture::createFlags texCreateFlags;
+	texCreateFlags.bTextureable = true;
```
Comment (DadSchoorse) [inline src/steamcompmgr.cpp:1019]: `vulkan_create_texture_from_bits` already does this.

**Pull request merged**